mediation_service/mediation/prepare_ssp_response.py

```diff
@@ -154,7 +168,49 @@ def __transform_allergy_local_references(ssp_response: dict):
                 ] = patient_list.get(patient_id[1], patient_ref)
 
 
-def __filter_non_allergy_intolerance(ssp_response: dict):
+def _extract_resolved_allergies(ssp_response: dict):
+    """
+    Select resolved allergy intolerance resources from Ended allergies list
+    Return them as a list of Bundle Entry objects
+    """
+    query = parse("`this`.entry[*].resource.resourceType")
```

Comment:
This block of code is repeated inside of two of these internal functions, is it worth factoring it into its own helper function that we can call within `prepare_ssp_response()` to parse the data into this nice Python Object format before we start filtering and extracting?